aw-sync/src/main.rs

```diff
@@ -139,60 +156,94 @@ fn main() -> Result<(), Box<dyn Error>> {
+    match opts.command.unwrap_or(Commands::Daemon {
+        start_date: None,
+        buckets: None,
+        sync_db: None,
+    }) {
         // Start daemon
-        Commands::Daemon {} => {
+        Commands::Daemon {
+            start_date,
+            buckets,
+            sync_db,
+        } => {
             info!("Starting daemon...");
-            daemon(&client)?;
-        }
-        // Perform basic sync
-        Commands::Sync { host } => {
-            // Pull
-            match host {
-                Some(hosts) => {
-                    for host in hosts.iter() {
-                        info!("Pulling from host: {}", host);
-                        sync_wrapper::pull(host, &client)?;
-                    }
-                }
-                None => {
-                    info!("Pulling from all hosts");
-                    sync_wrapper::pull_all(&client)?;
-                }
-            }
 
-            // Push
-            info!("Pushing local data");
-            sync_wrapper::push(&client)?
+            // Use an empty vector to sync all buckets for these cases:
+            // 1. When --buckets '*' is supplied
+            // 2. When no bucket argument is provided (default)
+            let effective_buckets = if buckets.as_deref() == Some("*") || buckets.is_none() {
+                Some(vec![])
+            } else if let Some(buckets_str) = buckets {
+                Some(buckets_str.split(',').map(|s| s.to_string()).collect())
+            } else {
+                None
+            };
+
+            daemon(&client, start_date, effective_buckets, sync_db)?;
         }
-        // Perform two-way sync
-        Commands::SyncAdvanced {
+        // Perform sync
+        Commands::Sync {
+            host,
             start_date,
             buckets,
             mode,
             sync_db,
         } => {
-            let sync_dir = dirs::get_sync_dir()?;
-            if let Some(db_path) = &sync_db {
-                info!("Using sync db: {}", &db_path.display());
+            // Use an empty vector to sync all buckets for these cases:
+            // 1. When --buckets '*' is supplied
+            // 2. When no bucket argument is provided (default)
+            let effective_buckets = if buckets.as_deref() == Some("*") || buckets.is_none() {
+                Some(vec![])
+            } else if let Some(buckets_str) = buckets {
+                Some(buckets_str.split(',').map(|s| s.to_string()).collect())
+            } else {
+                None
+            };
 
-                if !db_path.is_absolute() {
-                    Err("Sync db path must be absolute")?
-                }
-                if !db_path.starts_with(&sync_dir) {
-                    Err("Sync db path must be in sync directory")?
+            // If advanced options are provided, use advanced sync mode
+            if start_date.is_some() || effective_buckets.is_some() || sync_db.is_some() {
```

Comment:
I think (may be wrong) that this may never make `effective_buckets = None` as it will return a `Some` if buckets is None (which is the default) and thus advanced will always be used?

@ellipsis-dev plz review